### DISCUSSION TITLE
guide-R/analyses_avancees/analyse-factorielle

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#43] [Forum]
# guide-R/analyses_avancees/analyse-factorielle

https://larmarange.github.io/guide-R/analyses_avancees/analyse-factorielle.html

**Comment by SedzroEKLOU** [comment]
Bonjour Mr Larmarange,
J'aimerais classer des ménages d'une enquête selon leur niveau de richesse comme cela ce fait avec les scores sur l'axe 1 de l'ACM avec ade4 et ensuite divisé en quintile mais je ne sais plus comment continuer afin de retrouver les classes. Avez-vous un exemple de code à me proposer pour que je m'en inspire? Merci.

**Reply by larmarange** [reply]
Bonjour,

voici un example de code, à adapter à votre contexte :

```r
library(ade4)
library(tidyverse)

# Sélection des variables et réalisation d'une Analyse factorielle
# avec dudi.mix() si variables catégorielles et continues
# avec dudi.acm() si variables catgéorielles seulement

acm <- 
  jeu_de_donnees |>
  select(everything())|> # adapter selon le cas
  dudi.acm(scannf = FALSE)

# récupération du score sur l'axe 1
jeu_de_donnees$wealthness_score <- acm$li$Axis1

# création des quintiles
jeu_de_donnees <- 
  jeu_de_donnees |> 
  mutate(
    wealthness_quintiles = cut(
      wealthness_score, 
      breaks = c(-Inf, quantile(wealthness_score, probs = c(.2, .4, .6, .8)), Inf)
    )
    
# il faut vérifier ensuite l'ordre des quintiles (le premier correspond-il
# aux ménages riches ou pauvres) en croisant les quintiles avec les variables
# ayant servi à la construction du score
```

**Comment by Chairman-Juvencio** [comment]
Bonjour Larmarange,
J'aimerais savoir si vous avez un exemple de code pour faire l'analyse factorielle avec données mixte ? Merci beaucoup

**Reply by larmarange** [reply]
non pas ici, mais il s'agit de la même approche, mais vous remplacez `dudi.acm()` par `dudi.mix()`.

Je vous invite à lire https://larmarange.github.io/guide-R/analyses_avancees/analyse-factorielle.html qui est plus à jour.

Pour un exemple de code, regardez la documentation de ade4 : https://adeverse.github.io/ade4/reference/dudi.mix.html

Cdlt

**Comment by Tojo37** [comment]
Bonjour Mr Larmarange,
Je viens d'utiliser R version 4.5.1, et j'aimerais faire de l'analyse des correspondances multiples pour mon enquêtes socio-écologique sur la perception des changements paysager. Pourriez-vous me fournir un exemple de code dont je pourrais m'inspirer? Merci.
